src/class-wpml-elementor-translatable-nodes.php

```diff
@@ -74,7 +74,11 @@ public function get( $node_id, $element ) {
 				if ( isset( $node_data['integration-class'] ) ) {
 					foreach ( $this->get_integration_classes( $node_data ) as $class ) {
 						try {
-							$node    = new $class();
+							if ( $class instanceof \WPML_Elementor_Module_With_Items ) {
+								$node = $class;
+							} else {
+								$node = new $class();
+							}
```

Comment:
@strategio I'm not a huge fan of this sort of changes in existing API.

I do understand the reasons, but every time we do that, we make our API less deterministic.

Anyway, this is a month old PR: is it still valid?

Reply:
Yes this PR is still valid.

I understand your argument and with checking whether it is an instance of the expected class it stays deterministic. I tried hard to find a different approach, with anonymous classes, using static properties, etc. but haven't found a way to make it work other than this...

Reply:
Personnally, I don't think this solution is still valid. As an alternative, we could also allow to pass a factory, but it requires more changes I believe (and you won't be a huge fan of this either).

I'll leave you the final word.